README.md

```diff
@@ -1,177 +1,78 @@
-and sometimes even mandate a *single specific nightly* version: `rust-semverver` as of this writing
-needs `nightly-2022-08-03`. As `rustc` internals change,
-[constant work is required](https://github.com/rust-lang/rust-semverver/search?q=Rustup+to&type=commits)
-to keep compiler-based semver tools up to date and working. It's not clear that there is a path
-toward minimizing this maintenance burden, or a path toward getting these tools running
-on stable Rust.
-
-Meanwhile, rustdoc-based tools are limited by the information contained in the rustdoc JSON output,
-but have a clearer stable Rust story: the rustdoc JSON generation itself is unstable
-(though working with a much wider range of nightlies, and on
-[a path toward stabilization](https://github.com/rust-lang/rust/issues/76578)),
-but the checking of the JSON files can be done on stable Rust. There is interest in
+Notes:
+- If using it on a massive codebase (multiple hundreds of thousands of lines of Rust),
+  the queries may be a bit slow: there is some `O(n^2)` scaling for `n` items in a few places that
+  I haven't had time to optimize down to `O(n)` yet. Apologies! I have temporarily prioritized
+  features over speed, and the runtime will improve significantly with a small amount of extra work.
+- **No false positives**: Currently, all queries report constructive proof of semver violations:
+  there are no false positives. They always list a file name and line number for the baseline item
+  that could not be found in the new code.
+- **There are false negatives**: This tool is a work-in-progress, and cannot check all kinds of
+  semver violations yet. Just because it doesn't find any semver issues doesn't mean
+  they don't exist.
+
+## FAQ
+
+### Why `cargo-semver-checks` instead of ...?
+
+[rust semverver](https://github.com/rust-lang/rust-semverver) builds on top of
+rustc internals to build rlib's and compare their metadata.  This strips the
+code down to the basics for identifying changes.  However, is tightly coupled
+to specific nightly compiler versions and [takes work to stay in
+sync](https://github.com/rust-lang/rust-semverver/search?q=Rustup+to&type=commits).
+
+[cargo breaking](https://github.com/iomentum/cargo-breakinga) effectively runs
+`cargo expand` and re-parses the code using
+[`syn`](https://crates.io/crates/syn) which requires re-implementing large
+swaths of rust's semantics to then lint the API for changes.
+This is limited to the feature and target the crate was compiled for.
+
+`cargo-semver-checks` sources its data from rustdoc's json output.  While
+unstable, the rate of change is fairly low, reducing the churn in keeping up.
+The lints are also written ontop of the `trustfall` ["query everything"
+engine](https://github.com/obi1kenobi/trustfall), reducing the work for writing
+and maintaining lints.  Because of the extra data that rustdoc includes, some
+level of feature/target awareness might be able to be introduced.
```

Comment:
```suggestion
The lints are also written as queries for the `trustfall` ["query everything"
engine](https://github.com/obi1kenobi/trustfall), reducing the work for adding
and maintaining lints.  Because of the extra data that rustdoc includes, some
level of feature/target awareness might be able to be introduced.
```
Just trying to clarify a bit. English is harder than Rust :)

If you see a way to reword this so the first sentence doesn't both start and end with `lints`, that'd be great. I'm getting tripped up on it but I couldn't figure out a great way out of that.